src/geometry.h

```diff
@@ -234,33 +236,34 @@ struct Rectangle {
 	 * @param w Width of the rectangle.
 	 * @param h Height of the rectangle.
 	 */
-	Rectangle(typename PT::CoordType x, typename PT::CoordType y, SZ w, SZ h)
+	constexpr Rectangle(typename PT::CoordType x, typename PT::CoordType y, SZ w, SZ h)
+	:
+		base  (x, y),
+		width (w),
+		height(h)
 	{
-		this->base.x = x;
-		this->base.y = y;
-		this->width  = w;
-		this->height = h;
 	}
 
 	/**
 	 * Copy constructor.
 	 * @param rect %Rectangle to copy.
 	 */
 	template<typename PT2, typename SZ2>
-	Rectangle(const Rectangle<PT2, SZ2> &rect)
+	constexpr Rectangle(const Rectangle<PT2, SZ2> &rect)
+	:
+		base  (rect.base  ),
+		width (rect.width ),
+		height(rect.height)
```

Comment:
You are aware that maintaining vertical alignment does take a lot of effort?
Change one name, and the entire column needs fixing.

I do like vertical alignment, but these look a bit too simple to have much benefit to me.
I'd probably even stick lines 253 to 256 all behind 252, but that may be me :)